src/pages/PullRequestPage/Header/HeaderTeam/hooks/usePullHeadDataTeam.tsx

```diff
@@ -0,0 +1,168 @@
+import { useQuery } from '@tanstack/react-query'
+import { z } from 'zod'
+
+import {
+  RepoNotFoundErrorSchema,
+  RepoOwnerNotActivatedErrorSchema,
+} from 'services/repo'
+import Api from 'shared/api'
+import A from 'ui/A'
+
+const CoverageObjSchema = z.object({
+  percentCovered: z.number().nullable(),
+})
+
+const ComparisonSchema = z.object({
+  __typename: z.literal('Comparison'),
+  patchTotals: CoverageObjSchema.nullable(),
+})
+
+const CompareWithBaseSchema = z.discriminatedUnion('__typename', [
+  ComparisonSchema,
+])
```

Comment:
You'll need to add in the other union types or else this could error out the whole page. I have already made the schemas should be in `services/comparisons/schemas`